Code/LauncherUnified/Launcher.cpp

```diff
@@ -232,13 +233,17 @@ namespace
         // our frame time to be managed by AzGameFramework::GameApplication
         // instead, which probably isn't going to happen anytime soon given
         // how many things depend on the ITimer interface).
-        bool continueRunning = true;
         ISystem* system = gEnv ? gEnv->pSystem : nullptr;
-        while (continueRunning)
+        while (!gameApplication.WasExitMainLoopRequested())
         {
             // Pump the system event loop
             gameApplication.PumpSystemEventLoopUntilEmpty();
 
+            if (gameApplication.WasExitMainLoopRequested())
+            {
+                break;
+            }
+
```

Comment:
This seems kind of "dowhily" for me. i.e `do ... while`

What if we pumped the System Event loop at the end this block instead?

Reply:
Pumping the system event loop should be the first thing we do every frame. We could maybe pump it once before entering the loop, then at the end as you suggest, and the behavior would remain the same, but conceptually I think it's important for anyone looking at it to understand that it's the first thing to happen every frame, so I'd prefer leaving it as is.

Reply:
I'm not sure under what conditions the application can be requested to exit. If `Tick()` can also call `Application::ExitMainLoop()`, and that frees resources that `PumpSystemEventLoopUntilEmpty()` expects to be valid, I think we'd be right back in the same situation. So I was erring on the side of caution here.